src/lib/buffer.jl

```diff
@@ -1,11 +1,13 @@
-grad_mut(b::Buffer) = fill!(similar(b.data, Any), nothing)
-grad_mut(b::Buffer{T}) where T<:Number = fill!(similar(b.data, float(T)), 0)
+grad_mut(cx::Context, b::Buffer{T}, ::Type{S}=Union{}) where {T, S} =
+  _get!(() -> fill!(similar(b.data, Any), nothing), cache(cx), b)
```

Comment:
```suggestion
grad_mut(cx::Context, b::Buffer, ::Type) =
  _get!(() -> fill!(similar(b.data, Any), nothing), cache(cx), b)
```
Unless the forced specializations are helping somewhere.

Reply:
They're not, was just code symmetry, will do this (although it still needs `=Union{}` arg default)